File: app/src/main/java/com/example/ase_project_nutrismart/Retrofit/ApiInterface.java
package com.example.ase_project_nutrismart.Retrofit;

import com.example.ase_project_nutrismart.Model.SelectedGroceryModel;
import com.example.ase_project_nutrismart.Response.BmiResponse;
import com.example.ase_project_nutrismart.Response.ExpiryResponse;
import com.example.ase_project_nutrismart.Response.Grocery;
import com.example.ase_project_nutrismart.Response.LoginResponse;
import com.example.ase_project_nutrismart.Response.PurchasedGrocery;
import com.example.ase_project_nutrismart.Response.SelectedGrocery;
import com.example.ase_project_nutrismart.Response.SignupListResponse;
import com.example.ase_project_nutrismart.Response.SignupResponse;
import com.google.gson.internal.bind.ArrayTypeAdapter;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

public interface ApiInterface {
    @POST("/api/v1/auth/signin")
    Call<LoginResponse> login(@Body Map<String, String> param);

    @POST("/api/v1/auth/signup")
    Call<SignupResponse>signup(@Body Map<String,String> param);

    @GET("/api/v1/application/bmiCalc/{mass}/{height}/{age}/{gender}")
            Call<BmiResponse>bmi(@Path("mass")String mass, @Path("height")String height,  @Path("age")String age,  @Path("gender")String gender);


    @GET("/api/v1/auth/list")
    Call<List<SignupListResponse>>userList();


    @GET("api/v1/master/getFoods")
    Call<PurchasedGrocery>groceryList();


    @POST("/api/v1/application/selectedgrocery")
    Call<SelectedGrocery>sendSelectedGrocery(@Body SelectedGroceryModel model);

    @GET("api/v1/application/getExpiredItems")
    Call<ExpiryResponse>getExpiryList(@Query("userEmail")String userEmail);

}

File: app/src/main/java/com/example/ase_project_nutrismart/Retrofit/ApiInterface.kt
package com.example.ase_project_nutrismart.Retrofit

import okhttp3.logging.HttpLoggingInterceptor.setLevel
import okhttp3.OkHttpClient.Builder.addInterceptor
import okhttp3.OkHttpClient.Builder.build
import retrofit2.Retrofit
import okhttp3.logging.HttpLoggingInterceptor
import okhttp3.OkHttpClient
import com.example.ase_project_nutrismart.Retrofit.APIClient
import retrofit2.converter.gson.GsonConverterFactory
import com.example.ase_project_nutrismart.Response.LoginResponse
import com.example.ase_project_nutrismart.Response.SignupResponse
import com.example.ase_project_nutrismart.Response.BmiResponse
import com.example.ase_project_nutrismart.Response.SignupListResponse
import com.example.ase_project_nutrismart.Response.PurchasedGrocery
import com.example.ase_project_nutrismart.Model.SelectedGroceryModel
import com.example.ase_project_nutrismart.Response.SelectedGrocery
import com.example.ase_project_nutrismart.Response.ExpiryResponse
import retrofit2.Call
import retrofit2.http.*

interface ApiInterface {
    @POST("/api/v1/auth/signin")
    fun login(@Body param: Map<String?, String?>?): Call<LoginResponse?>?

    @POST("/api/v1/auth/signup")
    fun signup(@Body param: Map<String?, String?>?): Call<SignupResponse?>?

    @GET("/api/v1/application/bmiCalc/{mass}/{height}/{age}/{gender}")
    fun bmi(
        @Path("mass") mass: String?,
        @Path("height") height: String?,
        @Path("age") age: String?,
        @Path("gender") gender: String?
    ): Call<BmiResponse?>?

    @GET("/api/v1/auth/list")
    fun userList(): Call<List<SignupListResponse?>?>?

    @GET("api/v1/master/getFoods")
    fun groceryList(): Call<PurchasedGrocery?>?

    @POST("/api/v1/application/selectedgrocery")
    fun sendSelectedGrocery(@Body model: SelectedGroceryModel?): Call<SelectedGrocery?>?

    @GET("api/v1/application/getExpiredItems")
    fun getExpiryList(@Query("userEmail") userEmail: String?): Call<ExpiryResponse?>?
}